client/asset/interface.go

```diff
@@ -326,6 +336,15 @@ type Rescanner interface {
 	Rescan(ctx context.Context) error
 }
 
+// Recoverer is a wallet implementation with recover functionality.
+type Recoverer interface {
+	// GetRecoveryCfg returns information that will help the wallet get
+	// back to its previous state after it is recreated.
+	GetRecoveryCfg() (map[string]string, error)
+	// Destroy will delete all the wallet files so the wallet can be recreated.
+	Destroy() error
```

Comment:
I wish that `Destroy` wasn't needed. `*Core` should really just delete the `DataDir`. We can't though, because `*Core` doesn't set the network (mainnet/testnet/simnet) directory and relies on the wallets to do that part. Dang.

Reply:
I suggest we change this to `Move()` at least, I think that is a better practice.

Reply:
Updated it to `Move()`.